### PR TITLE
#952 feature: add sealer exec command

### PR DESCRIPTION
**Pull request by zsyaoo**
#952 feature: add sealer exec command

### CODE REVIEW

File: sealer/cmd/exec.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		cluster, err := utils.GetClusterFromFile(clusterFile)
+		if err != nil {
+			return err
+		}
+		ipList := append(cluster.GetMasterIPList(), cluster.GetNodeIPList()...)
+		var wg sync.WaitGroup
+		for _, ip := range ipList {
+			sshCli, err := ssh.GetHostSSHClient(ip, cluster)
+			if err != nil {
+				return err
```
Comment (fanux) [inline sealer/cmd/exec.go:56]:
```
	Google := func(ctx context.Context, query string) ([]Result, error) {
		g, ctx := errgroup.WithContext(ctx)

		searches := []Search{Web, Image, Video}
		results := make([]Result, len(searches))
		for i, search := range searches {
			i, search := i, search // https://golang.org/doc/faq#closures_and_goroutines
			g.Go(func() error {
				result, err := search(ctx, query)
				if err == nil {
					results[i] = result
				}
				return err
			})
		}
		if err := g.Wait(); err != nil {
			return nil, err
		}
		return results, nil
	}
```

Using errgroup https://pkg.go.dev/golang.org/x/sync/errgroup.

And you can put then into a module named exec.

File: sealer/cmd/exec.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+specify the cluster name(If there is only one cluster in the $HOME/.sealer directory, it should be applied. ):
+    sealer exec -c my-cluster 'cat /etc/hosts'
+set role label to exec cmd:
+    sealer exec -c my-cluster -r master,slave,node1 'cat /etc/hosts'		
+`,
+	Args: cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		execCmd, err := exec.NewExecCmd(clusterName, roles)
+		if err != nil {
+			return nil
```
Comment (fanux) [inline sealer/cmd/exec.go:40]: return err

File: sealer/cmd/exec.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		}
+		return execCmd.RunCmd(args...)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(execCmd)
+	execCmd.Flags().StringVarP(&clusterName, "cluster-name", "c", "", "submit one cluster name")
+	execCmd.Flags().StringVarP(&roles, "roles", "r", "", "set role label to roles")
```
Comment (fanux) [inline sealer/cmd/exec.go:49]: roles should be string slice.

Comment (zsyaoo) [reply sealer/cmd/exec.go:49]: we can use  `sealer exec -c my-cluster -r master,node,test1 'cat /etc/hosts'  `

File: sealer/cmd/exec.go (hunk 2 above)
Comment (zsyaoo) [reply sealer/cmd/exec.go]: i fixed

File: sealer/cmd/exec.go (hunk 3 above)
Comment (zsyaoo) [reply sealer/cmd/exec.go:49]: > roles should be string slice.
we can use sealer exec -c my-cluster -r master,node,test1 'cat /etc/hosts'

Comment (zsyaoo) [reply sealer/cmd/exec.go:49]: we can use `sealer exec -c my-cluster -r master,node,test1 'cat /etc/hosts'`

**Review by fanux — APPROVED** [review]

**Pull request merged**